'const int' is not const enough for arraysize before C99, see
http://sourceware.org/bugzilla/show_bug.cgi?id=5160

--- bfd/elflink.c	2007-10-22 22:14:29.000000000 +0200
+++ bfd/elflink.c	2007-10-22 22:36:09.000000000 +0200
@@ -7387,19 +7387,18 @@
 	     size_t                        locsymcount,
 	     int                           signed_p)
 {
-  int           len;
-  int           symlen;
+  size_t len;
+  size_t symlen;
   bfd_vma       a;
   bfd_vma       b;
-  const int     bufsz = 4096;
-  char          symbuf [bufsz];
+  char          symbuf [4096];
   const char *  symend;
   bfd_boolean   symbol_is_section = FALSE;
 
   len = strlen (sym);
   symend = sym + len;
 
-  if (len < 1 || len > bufsz)
+  if (len < 1 || len > sizeof (symbuf))
     {
       bfd_set_error (bfd_error_invalid_operation);
       return FALSE;
@@ -7424,7 +7423,7 @@
       symlen = strtol (sym, &sym, 10);
       ++ sym; /* Skip the trailing ':'.  */
 
-      if ((symend < sym) || ((symlen + 1) > bufsz))
+      if ((symend < sym) || ((symlen + 1) > sizeof(symbuf)))
 	{
 	  bfd_set_error (bfd_error_invalid_operation);
 	  return FALSE;
